### ISSUE TITLE
GGUI Not Available

### ISSUE THREAD

**Issue by AmosChenZixuan** [taichi-dev/taichi#2887]
**Describe the bug**
Instantiating the new GUI window will catch a no-ggui error. Ran on Ubuntu 20.04 with NVIDIA GTX 1060 graphic card. Suspect some dependencies are not installed?

**To Reproduce**
Run any of the ggui example, or any code that contains this line:

```py
window = ti.ui.Window(name, res, vsync=True)
```

**Log/Screenshots**

```
$ ti example mass_spring_3d_ggui
[Taichi] version 0.7.31, llvm 10.0.0, commit fa68168a, linux, python 3.8.8

*******************************************
**      Taichi Programming Language      **
*******************************************

Docs:   https://taichi.rtfd.io/en/stable
GitHub: https://github.com/taichi-dev/taichi
Forum:  https://forum.taichi.graphics

Running example mass_spring_3d_ggui ...
[Taichi] Starting on arch=cuda
Traceback (most recent call last):
  File "/home/amos/Programs/anaconda3/envs/taichi/bin/ti", line 8, in <module>
    sys.exit(main())
  File "/home/amos/Programs/anaconda3/envs/taichi/lib/python3.8/site-packages/taichi/main.py", line 1083, in main
    return cli()
  File "/home/amos/Programs/anaconda3/envs/taichi/lib/python3.8/site-packages/taichi/main.py", line 28, in wrapper
    result = func(*args, **kwargs)
  File "/home/amos/Programs/anaconda3/envs/taichi/lib/python3.8/site-packages/taichi/main.py", line 89, in __call__
    return getattr(self, args.command)(sys.argv[2:])
  File "/home/amos/Programs/anaconda3/envs/taichi/lib/python3.8/site-packages/taichi/main.py", line 220, in example
    runpy.run_path(target, run_name='__main__')
  File "/home/amos/Programs/anaconda3/envs/taichi/lib/python3.8/runpy.py", line 265, in run_path
    return _run_module_code(code, init_globals, run_name,
  File "/home/amos/Programs/anaconda3/envs/taichi/lib/python3.8/runpy.py", line 97, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/home/amos/Programs/anaconda3/envs/taichi/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/amos/Programs/anaconda3/envs/taichi/lib/python3.8/site-packages/taichi/examples/ggui_examples/mass_spring_3d_ggui.py", line 97, in <module>
    window = ti.ui.Window("Cloth", (800, 800), vsync=True)
  File "/home/amos/Programs/anaconda3/envs/taichi/lib/python3.8/site-packages/taichi/ui/ui.py", line 30, in __init__
    err_no_ggui()
  File "/home/amos/Programs/anaconda3/envs/taichi/lib/python3.8/site-packages/taichi/ui/ui.py", line 26, in err_no_ggui
    raise Exception("GGUI Not Available")
Exception: GGUI Not Available
```

**Additional comments**
```
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 465.19.01    Driver Version: 465.19.01    CUDA Version: 11.3     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  NVIDIA GeForce ...  On   | 00000000:01:00.0  On |                  N/A |
| 46%   43C    P0    26W / 120W |    384MiB /  6070MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|    0   N/A  N/A      1052      G   /usr/lib/xorg/Xorg                259MiB |
|    0   N/A  N/A      1392      G   xfwm4                               2MiB |
|    0   N/A  N/A      2383      G   ...AAAAAAAAA= --shared-files       79MiB |
|    0   N/A  N/A      3643      G   ...AAAAAAAAA= --shared-files       39MiB |
+-----------------------------------------------------------------------------+

[Taichi] version 0.7.31, llvm 10.0.0, commit fa68168a, linux, python 3.8.8
```

**Comment by bobcao3** [comment]
I'm not 100% sure but I think GGUI is still not yet fully in the latest
release. If you built from source then TI_WITH_VULKAN flag needs to be on.

**Comment by AmosChenZixuan** [comment]
You might be right. I presumed the ggui is released only because the examples and docs are already out there. Also, I am super excited for this feature.

**Comment by AmesingFlank** [comment]
@bobcao3  is correct. We haven't fully released GGUI yet.
If you wish to experiment with it, set `TI_WITH_VULKAN` to ON in ur `cmake/TaichiCore.cmake`, and rebuild taichi from source.
Thanks for your support and patience.

**Comment by erizmr** [comment]
Assume you are working on linux, to use GGUI, there are several steps to do:
1. Install Vulkan
2. git clone the lastest taichi repo
3. Turn the flag `TI_WITH_VULKAN`  in cmake/TaichiCore.cmake to be on as @bobcao3  mentioned
3. Remove the `build` folder if it exists 
4. Re-build taichi with `python3 setup.py develop --user`

**Comment by k-ye** [comment]
Merged into #2883
